repositories/employeeRepository.go

```diff
@@ -16,71 +16,317 @@ type EmployeeRepository struct {
+	employeePers.Id.Set(id)
+
+	commandTag, err := employeePers.DeleteTx(&tx)
 	if err != nil {
 		return err
 	}
-	if commandTag.RowsAffected() != 1 {
+	if commandTag != 1 {
 		return errors.New("No row found to delete")
 	}
+
+	tx.Commit(context.Background())
 	return nil
 }
+
+// GetEmployeeExternalPermissions .
+func (repository *EmployeeRepository) GetEmployeeExternalPermissions(idReceivingCompany string, product models.Product) (models.ExternalRights, error) {
+	var allRights []models.ExternalRights
+	var rights models.ExternalRights
+
+	// 1. Using idReceivingCompany and idSharingCompany, acquire all external access rules for these two companies
+	tx, err := repository.DB.Begin(context.Background())
+	if err != nil {
+		return rights, err
+	}
+
+	defer tx.Rollback(context.Background())
+
+	queryExternalAccess := "SELECT * FROM external_access_rights WHERE idrc = $1 AND idsc = $2;"
+	rows, err := tx.Query(context.Background(), queryExternalAccess, idReceivingCompany, product.IDC)
+
+	if err != nil {
+		return rights, err
+	}
+	for rows.Next() {
+		var right models.ExternalRights
+		err := rows.Scan(&right.ID, &right.IDSC, &right.IDRC, &right.Read, &right.Update, &right.Delete, &right.Approved)
+		if err != nil {
+			return rights, err
+		}
+		allRights = append(allRights, right)
+	}
+	rows.Close()
+
+	if len(allRights) == 0 {
+		// 2a. If there is no rows returned employees from this company can't interact with this product
+		return rights, errors.New("You don't have any permission for this product")
+	} else if len(allRights) == 1 {
+		// 2b. If there is only 1 row returned, it means there are no constraints and we can return safely
+		rights = allRights[0]
+	} else {
+		// 2c. Otherwise, we need to acquire constraints, using ID of all constraints
+		for _, right := range allRights {
+			var accessConstraint models.AccessConstraint
+			err := tx.QueryRow(context.Background(), "select * from access_constraints where idear=$1", right.ID).
+				Scan(&accessConstraint.ID, &accessConstraint.IDEAR, &accessConstraint.OperatorID, &accessConstraint.PropertyID, &accessConstraint.PropertyValue)
+			if err != nil {
+				return rights, err
+			}
+			if checkConstraint(accessConstraint, product) && right.Approved {
+				rights = right
+			}
+		}
+	}
+
+	err = tx.Commit(context.Background())
+
+	if err != nil {
+		return rights, err
+	}
+
+	return rights, nil
+}
+
+// CheckCompaniesSharingEmployeeData .
+func (repository *EmployeeRepository) CheckCompaniesSharingEmployeeData(idReceivingCompany string, idSharingCompany string) (bool, error) {
+	var allRights []models.ExternalRights
+
+	// 1. Using idReceivingCompany and idSharingCompany, acquire all external access rules for these two companies
+	tx, err := repository.DB.Begin(context.Background())
+	if err != nil {
+		return false, err
+	}
+
+	defer tx.Rollback(context.Background())
+
+	queryExternalAccess := "SELECT * FROM external_access_rights WHERE idrc = $1 AND idsc = $2;"
```

Comment:
You could add to the query: AND approved = $3 and pass the TRUE argument. That way you would get a much more limited set of data and avoid unnecessary check loops, like:
	rightsApproved := false
	for _, right := range allRights {
		if right.Approved {
			rightsApproved = true
			break
		}
	}